basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+            lanelet::BasicLineString2d downsampled_start_centerline;
+            downsampled_start_centerline.reserve(400);
+            downsampled_start_centerline = carma_utils::containers::downsample_vector(starting_centerline, 5);
+            std::vector<lanelet::BasicPoint2d> start_centerline;
+            for(int i = 0;i<starting_centerline.size();++i){
+                start_centerline.push_back(starting_centerline[i]);
+            }
+
+            lanelet::BasicLineString2d downsampled_end_centerline;
+            downsampled_end_centerline.reserve(400);
+            downsampled_end_centerline = carma_utils::containers::downsample_vector(ending_centerline, 5);
+            std::vector<lanelet::BasicPoint2d> end_centerline;
+            for(int i = 0; i<ending_centerline.size();++i){
+                end_centerline.push_back(ending_centerline[i]);
+            }
+            //If points are not the same size - ensure same size along both centerlines
+            if(start_centerline.size() != end_centerline.size())
+            {
+                //Resample centerlines to same length
+                auto centerlines = resample_linestrings_to_same_length(start_centerline, end_centerline);
+                start_centerline = centerlines.front();
+                end_centerline = centerlines.back();
+
+            }
+
+            //Create Trajectory geometry
+            double delta_step = 1.0 / start_centerline.size();
+
+            for (int i = 0; i < start_centerline.size(); i++)
+            {
+                lanelet::BasicPoint2d current_position;
+                lanelet::BasicPoint2d start_lane_pt = start_centerline[i];
+                lanelet::BasicPoint2d end_lane_pt = end_centerline[i];
+                double delta = delta_step * i;
+                current_position.x() = end_lane_pt.x() * delta + (1 - delta) * start_lane_pt.x();
+                current_position.y() = end_lane_pt.y() * delta + (1 - delta) * start_lane_pt.y();
+
+                centerline_points.push_back(current_position);
+            }
+
+            //Add points from following lanelet to provide sufficient distance for adding buffer
+            auto following_lanelets = wm->getMapRoutingGraph()->following(ending_lanelet, false);
+            if(!following_lanelets.empty()){
+            //Arbitrarily choosing first following lanelet for buffer since points are only being used to fit spline
+            auto following_lanelet_centerline = following_lanelets.front().centerline2d().basicLineString();
+            centerline_points.insert(centerline_points.end(), following_lanelet_centerline.begin(), 
+                                                                        following_lanelet_centerline.end());
+            }
+
+            return centerline_points;
+        }
+
+        std::vector<std::vector<lanelet::BasicPoint2d>> resample_linestrings_to_same_length(std::vector<lanelet::BasicPoint2d> line_1, std::vector<lanelet::BasicPoint2d> line_2){
```

Comment:
This is a pretty costly function since both the spline fitting occurs twice and the compute_arc_lengths method is called twice. I think we ought to log the execution time as a debug log.